### ISSUE TITLE
no whitespace generated if link appears at start of line in the source

### ISSUE THREAD

**Issue by holtzermann17** [theiceshelf/firn#82]
**Example  Org Mode text:**

```org
In this meeting we did a live demo and source code exploration of
[[https://code.librehq.com/qhong/crdt.el][crdt.el]] with the author, Qiantan Hong.  In the last 15 minutes we
outlined some plans for the next meeting, to focus on Patterns + Org
mode.
```

**Produces HTML:**

```html
<p><span>In this meeting we did a live demo and source code exploration of</span><a class="firn-external" href="https://code.librehq.com/qhong/crdt.el" target="_blank">crdt.el</a><span> with the author, Qiantan Hong.  In the last 15 minutes we
outlined some plans for the next meeting, to focus on Patterns + Org
mode.</span></p>
```

**Reads as follows:**

> In this meeting we did a live demo and source code exploration ofcrdt.el with the author, Qiantan Hong. In the last 15 minutes we outlined some plans for the next meeting, to focus on Patterns + Org mode.

I'm not sure why the text segments are inside `<span>`... but more importantly, there is no space in front of "crdt.el".  The current work around is to make sure that links don't appear at the start of a line, but it can be easy to forget to check that.

**Comment by teesloane** [comment]
This is a small bug that has been annoying me whenever I run into it. The parser sees line breaks as separates entries in a paragraph:

```
{
  "type": "document",
  "children": [
    {
      "type": "section",
      "children": [
        {
          "type": "paragraph",
          "children": [
            {
              "type": "text",
              "value": "In this meeting we did a live demo and source code exploration of\r\n"
            },
            {
              "type": "link",
              "path": "https://code.librehq.com/qhong/crdt.el",
              "desc": "crdt.el"
            },
            {
              "type": "text",
              "value": " with the "
            },
            {
              "type": "bold",
              "children": [
                {
                  "type": "bold",
                  "children": [
                    {
                      "type": "text",
                      "value": "author"
                    }
                  ]
                }
              ]
            },
            {
              "type": "text",
              "value": ", Qiantan Hong.  In the last 15 minutes we\r\noutlined some plans for the next meeting, to focus on Patterns + Org\r\nmode.\r"
            }
          ]
        }
      ]
    }
  ]
}
```

Perhaps `to-html` could be updated to remove the span wrapping of plain text and then we could flatten it all under the [:p ...] content it would belong too... I will have to play around in the repl a bit soon to see what I can come up with. I have been avoiding this bug by just now starting new lines with links as well 🤦

**Comment by teesloane** [comment]
Fixed by 0.15+
